pyccel/codegen/printing/fcode.py

```diff
@@ -560,6 +560,23 @@ def _build_gFTL_module(self, expr_type):
                     raise NotImplementedError("Support for lists of types defined in other modules is not yet implemented")
                 macros.append(MacroDefinition('Vector', expr_type))
                 macros.append(MacroDefinition('VectorIterator', IteratorType(expr_type)))
+            elif isinstance(expr_type, HomogeneousSetType):
+                include = Import(LiteralString('set/template.inc'), Module('_', (), ()))
+                element_type = expr_type.element_type
+                if isinstance(element_type, FixedSizeNumericType):
+                    tmpVar_x = Variable(element_type, 'x')
+                    tmpVar_y = Variable(element_type, 'y')
+                    if isinstance(element_type.primitive_type, PrimitiveComplexType):
+                        lt_def = PyccelAssociativeParenthesis(PyccelLt(NumpyAbs(tmpVar_x), NumpyAbs(tmpVar_y)))
```

Comment:
The modulus is not enough to define a total order on complex numbers
the following code should output 2 but it outputs 1:
```python
if __name__ == '__main__':
    s = {0+1j , 1+0j }
    print(len(s))
```
the real and imaginary parts should also be compared

Reply:
Hmm, good point. Maybe I need to create a function for this. I tried `(x.real == y.real ? x.imag < y.imag : x.real < y.real)` but when the macro was expanded it ran over the maximum line length

Reply:
I have changed the test so it would have failed before my changes [1abdf73](https://github.com/pyccel/pyccel/pull/2004/commits/1abdf7307507c929f836081946a0224c7ec0f46f)

Reply:
The code now uses a function

Reply:
Well done!